spec/features/consumer/shopping/checkout_spec.rb

```diff
@@ -93,365 +115,339 @@
+        order.distributor.allow_order_changes = true
+        order.distributor.save
+        visit checkout_path
+      end
 
-            place_order
-            expect(page).to have_content "Your order has been processed successfully"
+      it "informs about previous orders" do
+        expect(page).to have_content("You have an order for this order cycle already.")
+      end
+    end
 
-            expect(order.reload.bill_address.address1).to eq '123 Your Head'
-            expect(order.reload.ship_address.address1).to eq '123 Your Head'
+    context "when the user has a preset shipping and billing address" do
+      before do
+        user.bill_address = build(:address)
+        user.ship_address = build(:address)
+        user.save!
+      end
 
-            expect(order.customer.bill_address.address1).to eq '123 Your Head'
-            expect(order.customer.ship_address.address1).to eq '123 Your Head'
+      it "checks out successfully" do
+        visit checkout_path
+        choose shipping_with_fee.name
+        choose check_without_fee.name
 
-            expect(user.reload.bill_address.address1).to eq '123 Your Head'
-            expect(user.reload.ship_address.address1).to eq '123 Your Head'
-          end
-        end
+        expect do
+          place_order
+          expect(page).to have_content "Your order has been processed successfully"
+        end.to enqueue_job ConfirmOrderJob
+      end
+    end
 
-        it "it doesn't tell about previous orders" do
-          expect(page).to have_no_content("You have an order for this order cycle already.")
-        end
+    context "with Stripe" do
+      let!(:stripe_pm) do
+        create(:stripe_payment_method, distributors: [distributor])
       end
 
-      context "with previous orders" do
-        let!(:prev_order) { create(:completed_order_with_totals, order_cycle: order_cycle, distributor: distributor, user: order.user) }
+      let!(:saved_card) do
+        create(:credit_card,
+               user_id: user.id,
+               month: "01",
+               year: "2025",
+               cc_type: "visa",
+               number: "1111111111111111",
+               payment_method_id: stripe_pm.id,
+               gateway_customer_profile_id: "i_am_saved")
+      end
 
-        before do
-          order.distributor.allow_order_changes = true
-          order.distributor.save
-          visit checkout_path
-        end
+      let!(:stripe_account) { create(:stripe_account, enterprise_id: distributor.id, stripe_user_id: 'some_id') }
 
-        it "informs about previous orders" do
-          expect(page).to have_content("You have an order for this order cycle already.")
-        end
+      let(:response_mock) { { id: "ch_1234", object: "charge", amount: 2000 } }
+
+      before do
+        allow(Stripe).to receive(:api_key) { "sk_test_12345" }
+        allow(Stripe).to receive(:publishable_key) { "some_key" }
+        Spree::Config.set(stripe_connect_enabled: true)
```

Comment:
if we're not restoring the state and this is not affecting any other specs, we could as well make this purposeful and explicit and set it globally in the spec_helper.

Reply:
the default is false, I dont think we need to set it in spec_helper.
I added a commit to add restore original value in all specs that change this value.
I think this will make everything faster as stripe will always be disabled except when explicitly enabled.

Reply:
please have a look at the new commit.